tasks/insights-client.yml

```
@@ -10,6 +10,26 @@
   when:
     - rhc_state | d("present") != "reconnect"
 
+- name: Add insights tags
+  copy:
+    dest: /etc/insights-client/tags.yml
+    content: "{{ rhc_insights.tags | to_nice_yaml(indent=2) }}"
+    mode: '0644'
+  register: __insights_tags_added
+  when:
+    - rhc_insights.tags is defined
+    - rhc_insights.tags.state | d(omit) != "absent"
+    - rhc_insights.tags | d({}) | length > 0
+
+- name: Delete insights tags
+  file:
+    state: absent
+    path: /etc/insights-client/tags.yml
+  register: __insights_tags_removed
+  when:
```

```suggestion
  when: rhc_insights.tags | d(None) == {"state": "absent"}
```
I think that's the only condition you need - only if rhc_insights.tags is defined and equal to `{"state": "absent"}` will the tags be removed and the file deleted

---

Or, instead of hardcoding `{"state": "absent"}`, use the already-made `__rhc_state_absent` variable ;)